### PR TITLE
Create a better strategy for RGT_ACCT_ID and RGT_SUBMIT_QUEUE

### LINKED ISSUE #24: Replacements variable `project_id` is overriding RGT_ACCT_ID

### PR DESCRIPTION
**Pull request by hagertnl**
This PR changes the 3 schedulers (lsf.py, slurm.py, pbs.py) to have the following hierarchy when setting the submission project and queue:

For project:
(lowest priority)
- acct_id from <machine>.ini -- this propagates as the RGT_ACCT_ID in the environment** -- I think this should probably be `project_id`, if we want to match what batch queue does.
- project_id from <app>/<test>/Scripts/rgt_test_input.ini -- this overrides the RGT_PROJECT_ID in the environment
- RGT_ACCT_ID -- set by the user. If we used RGT_PROJECT_ID, the test would still override this.
(highest priority)

For queue:
(lowest priority)
- submit_queue from <machine>.ini -- this propagates as the RGT_SUBMIT_QUEUE in the environment
- batch_queue from <app>/<test>/Scripts/rgt_test_input.ini -- this overrides RGT_SUBMIT_QUEUE in the environment
- RGT_BATCH_QUEUE -- must be set by the user. If we used RGT_SUBMIT_QUEUE, the test would still override this
(highest priority)

This depends on the user using this correctly. So for example, we have `acct_id` in most of our machine.ini files right now, but I believe this should probably be `project_id`. Under the new priority, I think that because `acct_id` is setting `RGT_ACCT_ID` in the environment, the test's `project_id` would always end up being overwritten.

In summary: by default, RGT_BATCH_QUEUE and RGT_ACCT_ID should not be set by the harness. These should be the variables the user changes if they want to force submission to a specific queue/project.

### CODE REVIEW

**Comment by hagertnl** [comment]
Tested and I confirm that if `acct_id` is set in <machine>.ini, this setup doesn't quite work as expected. In that case, the test's ini file has the lowest priority (to the point that it's actually always ignored).

But, if `acct_id` is changed to `project_id` in <machine>.ini, then the hierarchy works correctly and I am able to successfully override the environment variables for account and batch queue using RGT_BATCH_QUEUE and RGT_ACCT_ID.

**Review by ddietz89 — COMMENTED** [review]
The code changes look fine to me. Should we have @verolero86 weigh on the changes too?

**Review by verolero86 — APPROVED** [review]
these look good to me 👍 ! we should also update the intended usage in the docs to make it clear when each one should / should not be used.

**Comment by hagertnl** [comment]
I think I may have this backwards -- I think that the *user* should specify `RGT_SUBMIT_QUEUE`, so that it is consistent with the `RGT_SUBMIT_ARGS` environment variable, and we should have the harness use `RGT_BATCH_QUEUE` internally.

Thoughts?

Definitely will update the docs to reflect whichever usage is preferred, I just thought having it be `RGT_SUBMIT_<x>` might be easier.

**Comment by hagertnl** [comment]
Converted to draft. Implementing the following hierarchy (lowest to highest)

1. batch_queue and project_id from machine ini
2. batch_queue and project_id from test ini
3. RGT_SUBMIT_QUEUE and RGT_ACCT_ID set by user

This is a change from current. Currently, machine.ini files specify `submit_queue`, not `batch_queue`. To be more consistent with parameters like `RGT_SUBMIT_ARGS`, I thought changing the user-override parameter to `RGT_SUBMIT_QUEUE` made sense. The account behavior shouldn't change, I believe that's what we already have in place.

If an old config file is used with this new precedence, the result will simply be that the machine.ini file will always have priority when setting batch queue and account ID (since submit_queue becomes RGT_SUBMIT_QUEUE and acct_id becomes RGT_ACCT_ID, which will have the highest precedence).

**Comment by hagertnl** [comment]
New hierarchy is now in place. RGT_SUBMIT_QUEUE and RGT_ACCT_ID have authority over machine.ini and rgt_test_input.ini. The example configs have been updated to reflect the latest changes to the docs and to this hierarchy.

One note that I'll make is the placement of `project_id` in the config file. It was previously in TestshotDefaults, but I moved this to be in MachineDetails with batch_queue and submit_args. Since we tend to talk about batch_queue and project_id together, I thought this would be a good change, but I am open to discussion about this. There's no impact to functionality.

**Comment by hagertnl** [comment]
If we go with this method, should we consider also changing `RGT_ACCT_ID` to `RGT_SUBMIT_ACCT`? If we're trying to standardize to the `RGT_SUBMIT_*` format. Will update the docs for whatever we decide.

**Comment by hagertnl** [comment]
As of the latest commit, we now have the following precedence, from lowest to highest:
- batch_queue, project_id from machine.ini
- batch_queue, project_id from rgt_test_input.ini
- RGT_SUBMIT_QUEUE, RGT_SUBMIT_ACCT from user's environment

These changes will be documented.
